app/lib/actions/katello/upstream_subscriptions/remove_entitlement.rb

```diff
@@ -0,0 +1,19 @@
+module Actions
+  module Katello
+    module UpstreamSubscriptions
+      class RemoveEntitlement < Actions::Base
+        middleware.use ::Actions::Middleware::PropagateCandlepinErrors
```

Comment:
This action locks the org if we raise an error when, for example, the entitlement was already removed from the allocation

The user doesn't really have a way to work around that, so the locking doesn't make sense. Should the action be cancellable or something similar?

Reply:
I think this is taken care of with the `rescue_strategy`: https://github.com/Katello/katello/pull/7235/files#diff-562f4c0f3f476d7dd66af859c597ce58R24

That should keep the task from locking on failure.

Reply:
cool, I think I'll use that

Reply:
@jturel Still need to follow up here.

Reply:
I added it in the 'parent' action ::BindEntitlements

Reply:
Should this `middleware.use` be in the parent action, or this one, and why?

Reply:
This lower-level action is the one that's hitting candlepin, so I want it to propagate the error - and it appears that it bubbles up through the parent action too.